apps/vmq_diversity/src/vmq_diversity_sup.erl

```diff
@@ -108,12 +108,14 @@ start_all_pools([{mongodb, ProviderConfig}|Rest], Acc) ->
     PoolOpts = proplists:get_value(opts, ProviderConfig, []),
     Size = proplists:get_value(size, PoolOpts, 5),
     MaxOverflow = proplists:get_value(max_overflow, ProviderConfig, 20),
-    WorkerArgs = lists:keydelete(size, 1,
-                                 lists:keydelete(max_overflow, 1, PoolOpts)),
+    WorkerArgs0 = lists:keydelete(size, 1,
+                                  lists:keydelete(max_overflow, 1, PoolOpts)),
     StartFun = fun() ->
-                       Ssl = proplists:get_value(ssl, WorkerArgs, false),
-                       SslOpts = proplists:get_value(ssl_opts, WorkerArgs),
-                       mc_worker:start_link([{ssl,Ssl}, {ssl_opts, SslOpts}|WorkerArgs])
+                       Ssl = proplists:get_value(ssl, WorkerArgs0, false),
+                       SslOpts = proplists:get_value(ssl_opts, WorkerArgs0),
+                       WorkerArgs1 = proplists:delete(ssl, WorkerArgs0),
+                       WorkerArgs2 = proplists:delete(ssl_opts, WorkerArgs1),
+                       mc_worker_api:connect([{ssl,Ssl}, {ssl_opts, SslOpts}|WorkerArgs2])
```

Comment:
does `mc_worker_api:connect/1` change anything in behaviour? vs. `mc_worker:start_link/1`? 
(fine with me, I'm just wondering what the difference is as the params are the same.

Reply:
Yeah. It's a [thin wrapper](https://github.com/comtihon/mongodb-erlang/blob/master/src/api/mc_worker_api.erl#L59-L71) around `mc:worker:start_link/1`.

If you pass `login` and `password` in the proplist, it authenticates the connection.

https://github.com/comtihon/mongodb-erlang/blob/master/src/api/mc_worker_api.erl#L64-L68

I am not sure how authentication was working without this.